Feature/update docs module

pasqenr opened the pull request:
### Changelog:

* Add an `analysis_options.yaml` file to enable more lints
* Lint the example
* Add optional parameter `client` in `EncryptionHelper.getGovernmentRsaPubKey()`
* `SignHelper.signSorted` parameter `data` now is defined as `Map<String, dynamic>`
* In `CommercioDocHelper.fromWallet()` the parameter `encryptedData` now is a `Set`
* In `CommercioDocReceiptHelper.fromWallet()` the `proof` now is `null` by default
* In `DocsHelper.shareDocument()` the parameter `encryptedData` now is a `Set`
* Renamed typo in method `getSendDocuments` to `getSentDocuments` in `DocsHelper` 
* Add `http.Client client` parameter in `getSentDocuments()`, `getReceivedDocuments()`, `getSentReceipts()` and `getReceivedReceipts()`
* The parameter `encryptedData` now is a `Set` in the method `encryptField()`
* Remove the duplication of `EncryptedData` with `CommercioEncryptedData`  in `encryptField()`
* Update the `CommercioDoc`, `CommercioDocReceipt`, `MsgSendDocumentReceipt`,  entities to match the new specification
* Update the `Network` helper, removing the static `client` with optional parameters, let `queryChain()` to use the `query()` method instead of duplicate code
* Add tests and bump code coverage

### Note

I've set the merge branch to `feature/commerciokyc-module` because this branch has already all the other branch commits.

For more information about this PR see issue #28.

marcotradenet commented on lib/crypto/encryption_helper.dart:
```diff
@@ -4,27 +4,36 @@ import 'dart:typed_data';
 
     if (identityResponseRaw == null) {
-      throw FormatException('Cannot get government RSA public key');
+      throw const FormatException('Cannot get government RSA public key');
```
Should be 'Cannot get tumbler RSA public key'

pasqenr replied:
Thank you, fixed in 9a773a1.

marcotradenet commented on lib/utils/utils.dart:
```diff
@@ -44,23 +44,24 @@ StdFee calculateDefaultFee({
+/// Returns [true] if the provided [str] is a valid bech32 string
+/// (no checksum check is done).
 /// See: https://en.bitcoin.it/wiki/Bech32
 bool matchBech32Format(String str) {
```
This function doesn't match exactly bech32 format: some string like `----1---------------` can used.    
Anyway it's ok because the chain controls the format.     
Leave it and we will improve it before the version release.

pasqenr replied:
Thanks, I've missed that and many others, I suppose. When I started to work on the `docs` module I wanted to try porting the Python reference bech32 checksum, but it's not a trivial task (maybe with the Go implementation it's better) and left only a very basic regex as you see.

marcotradenet reviewed (approved):
Pr it's ok for me

zenit70 reviewed (commented):
`lib/entities/docs/commercio_doc.dart`

![immagine](https://user-images.githubusercontent.com/19747789/108326806-f6a0d500-71ca-11eb-8a54-3eb7fd630148.png)


@pasqenr In the `commercio/MsgShareDocument` documentation, there are some required and some not required fields: about the latter, some must be sent even if not filled in (i.e. checksum), while others must not be sent. 
In the `CommercioDoc` class at the beginning there is this declaration: `@JsonSerializable(explicitToJson: true, includeIfNull: false)`.
Does `includeIfNull: false` apply to all properties in the class including `checksum'?

zenit70 reviewed (commented):
`lib/entities/docs/commercio_doc.dart`

@pasqenr 
![immagine](https://user-images.githubusercontent.com/19747789/108331566-2eaa1700-71cf-11eb-94d5-5adfed4c0852.png)

Also the class `CommercioDoSign` has the declaration `@JsonSerializable(explicitToJson: true, includeIfNull: false)`, but the prop `sdn_data` must be sent with `null` value according to the documentation that i have attached.

pasqenr commented:
Thank you for your findings @zenit70, I've missed those fields and the ShareDoc messages were correctly accepted anyway on the chain. Now are correctly included even if those fields are `null`.

I've also added a check on the recipients DDOs before compose an encrypted ShareDoc message to provide a better explanation error to the developer and fix the too permissive old algorithm. 

As last thing I've linted some other files and fixed a couple of throws in `props` fields.
